### PR TITLE
feat(backend):generate APIs on the backend using openapi

### PR DESCRIPTION
**Pull request by nancymuyeh**
Generate APIs on the backend #15 based on https://openapi-generator.tech/docs/plugins

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,3 +56,25 @@ dependencyManagement {
 tasks.named('test') {
 	useJUnitPlatform()
 }
+
+openApiGenerate {
+  generatorName = 'spring'
+  inputSpec = "$rootDir/../lynx-scanner/openapi.yaml".toString()
```
Comment (stephane-segning) [inline build.gradle:62]: This path doesn't exist in the current project. Use the guy submodule already configured

File: build.gradle [hunk 2]
```diff
@@ -54,3 +56,25 @@ dependencyManagement {
...
+  outputDir = "$buildDir/generated".toString()
+  apiPackage = '${api-package}.api'
+  modelPackage = '${api-package}.model'
+  invokerPackage = '${api-package}.handler'
```
Comment (stephane-segning) [inline build.gradle:66]: This variable doesn't exist: `api-package`.

File: build.gradle [hunk 3]
```diff
@@ -54,3 +56,25 @@ dependencyManagement {
 tasks.named('test') {
 	useJUnitPlatform()
 }
+
+openApiGenerate {
+  generatorName = 'spring'
+  inputSpec = "$rootDir/../lynx-scanner/openapi.yaml".toString()
+  outputDir = "$buildDir/generated".toString()
```
Comment (stephane-segning) [inline build.gradle:63]: You don't need the `.toString()` here

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Good work! Did you btw tested if it works? Plus the path you provided make reference to lynx-frontend which doesn't exist. Update that to a path into the repo. Also try running it. It would fail because some variables cannot be resolved.

File: build.gradle (hunk 1 above)
Comment (nancymuyeh) [reply build.gradle]: Done

File: build.gradle (hunk 3 above)
Comment (nancymuyeh) [reply build.gradle]: remove toString()

**Pull request closed without merging**